hazelcast/src/main/java/com/hazelcast/jet/impl/operation/AsyncOperation.java

```diff
@@ -60,7 +61,9 @@ public final void run() {
         future.whenComplete(withTryCatch(getLogger(), (r, f) -> {
             if (f != null) {
                 Throwable peeledFailure = peel(f);
-                logError(peeledFailure);
+                if (!isTechnicalCancellationException(f)) {
+                    logError(peeledFailure);
```

Comment:
Perhaps instead of no logging at all, logging at FINEST level will be better.

Reply:
I have removed the `CancellationException` from the list, so that will be logged.

The `JobTerminateRequestedException`, `ResultLimitReachedException`, `TerminatedWithSnapshotException` are printed in the member logs earlier at debug level.

So I think currently it is fine to not log these 3 at any level as it would be duplicated.